graphql/server.go

```diff
@@ -165,14 +167,34 @@ func (c *conn) handleSubscribe(id string, subscribe *subscribeMessage) error {
 		ctx = batch.WithBatching(ctx)
 
 		start := time.Now()
-		span, ctx := opentracing.StartSpanFromContext(ctx, "thunder.subscription")
+
 		c.logger.StartExecution(ctx, tags, initial)
-		current, err := e.Execute(ctx, c.schema.Query, nil, query)
+
+		output := &ComputationOutput{
+			Metadata: make(map[string]interface{}),
+		}
+
+		var middlewares []MiddlewareFunc
+		middlewares = append(middlewares, c.middlewares...)
+		middlewares = append(middlewares, func(input *ComputationInput, output *ComputationOutput, next NextFunc) {
+			output.Current, output.Error = e.Execute(input.Ctx, c.schema.Query, nil, input.ParsedQuery)
+			next(input, output)
+		})
+
+		runMiddlewares(middlewares, &ComputationInput{
+			Ctx:         ctx,
+			Id:          id,
+			ParsedQuery: query,
+			Previous:    previous,
```

Comment:
I think it'd be useful for letting the middleware know whether or not we are doing a recomputation. In the tracing middleware, I have a line that looks like:
```golang
span.SetTag("computation.initial", input.Previous == nil)
```

If we were to also to model the graphql logger as middleware in the future, it'd also be useful to have this "is recomputation" bit.